spec/filters/grok_spec.rb

```diff
@@ -25,347 +20,353 @@ def self.sample(message, &block)
+          expect( event.get("tags") ).to be nil
+          expect( event.get(ecs_select[disabled: "[syslog5424_pri]", v1: "[log][syslog][priority]"]).to_s ).to eq "30"
+          expect( event.get ecs_select[disabled: "[syslog5424_ver]", v1: "[system][syslog][version]"] ).to eq "1"
+          expect( event.get ecs_select[disabled: "[syslog5424_ts]", v1: "[timestamp]"] ).to eq "2014-04-04T16:44:07+02:00"
+          expect( event.get ecs_select[disabled: "[syslog5424_host]", v1: "[host][hostname]"] ).to eq "osctrl01"
+          expect( event.get ecs_select[disabled: "[syslog5424_app]", v1: "[process][name]"] ).to eq "dnsmasq-dhcp"
+          expect( event.get(ecs_select[disabled: "[syslog5424_proc]", v1: "[process][pid]"] ).to_s ).to eq "8048"
+          expect( event.get ecs_select[disabled: "[syslog5424_msgid]", v1: "[log][syslog][msgid]"] ).to eq nil
+          expect( event.get ecs_select[disabled: "[syslog5424_sd]", v1: "[system][syslog][structured_data]"] ).to eq nil
+          expect( event.get ecs_select[disabled: "[syslog5424_msg]", v1: "[message][1]"] ).to eq "Appname contains a dash"
+        end
 
-    sample "hello 1234" do
-      expect( event.get("FIZZLE") ).to eql '1234'
+        sample "<30>1 2014-04-04T16:44:07+02:00 osctrl01 - 8048 - -  Appname is nil" do
+          expect( event.get("tags") ).to be nil
+          expect( event.get(ecs_select[disabled: "[syslog5424_pri]", v1: "[log][syslog][priority]"]).to_s ).to eq "30"
+          expect( event.get ecs_select[disabled: "[syslog5424_ver]", v1: "[system][syslog][version]"] ).to eq "1"
+          expect( event.get ecs_select[disabled: "[syslog5424_ts]", v1: "[timestamp]"] ).to eq "2014-04-04T16:44:07+02:00"
+          expect( event.get ecs_select[disabled: "[syslog5424_host]", v1: "[host][hostname]"] ).to eq "osctrl01"
+          expect( event.get ecs_select[disabled: "[syslog5424_app]", v1: "[process][name]"] ).to eq nil
+          expect( event.get(ecs_select[disabled: "[syslog5424_proc]", v1: "[process][pid]"] ).to_s ).to eq "8048"
+          expect( event.get ecs_select[disabled: "[syslog5424_msgid]", v1: "[log][syslog][msgid]"] ).to eq nil
+          expect( event.get ecs_select[disabled: "[syslog5424_sd]", v1: "[system][syslog][structured_data]"] ).to eq nil
+          expect( event.get ecs_select[disabled: "[syslog5424_msg]", v1: "[message][1]"] ).to eq "Appname is nil"
+        end
+      end
     end
   end
 
-  describe "processing selected fields" do
-    let(:config) {
-      {
-        'match' => { "message" => "%{WORD:word}", "examplefield" => "%{NUMBER:num}" },
-        'break_on_match' => false
-      }
-    }
-    let(:data) { { "message" => "hello world", "examplefield" => "12345" } }
+  describe "non ecs" do
+    before(:each) do
+      subject.register
+      subject.filter(event)
+    end
+
+    describe "parsing an event with multiple messages (array of strings)", if: false do
```

Comment:
Does it mean that's never executed?

Reply:
Confirmed that it does not execute. I have updated the test to make it runs